boot/bootutil/src/bootutil_misc.c

```diff
@@ -309,13 +309,13 @@ boot_write_trailer(const struct flash_area *fap, uint32_t off,
     int rc;
 
     align = flash_area_align(fap);
-    if (inlen > BOOT_MAX_ALIGN || align > BOOT_MAX_ALIGN) {
+    if (inlen > BOOT_MAX_ALIGN || align > BOOT_MAX_ALIGN ||
+	BOOT_MAX_ALIGN % align) {
         return -1;
     }
     erased_val = flash_area_erased_val(fap);
-    if (align < inlen) {
-        align = inlen;
-    }
+    align = ((inlen + align - 1) / align) * align;
```

Comment:
The division seems kind of heavy here, but I suppose this code doesn't run that frequently. We could ensure somewhere that the alignment is a power of two (it doesn't really make sense for it to not be), and then change this to `((inline + align - 1) & ~(align - 1)`, but in this instance, it is probably not worth doing that.

Reply:
Will change.